src/Icicle/Avalanche/Statement/Simp.hs

```diff
@@ -331,10 +334,11 @@ substXinS a_fresh name payload statements
 --  * Remove some other useless code:
 --      statements that do not update accumulators or return a value are silly.
 --
-thresher :: (Ord n, Eq p) => a -> Statement a n p -> FixT (Fresh n) (Statement a n p)
+thresher :: (Hashable n, Eq n, Eq p) => a -> Statement a n p -> FixT (Fresh n) (Statement a n p)
 thresher a_fresh statements
- = transformUDStmt trans emptyExpEnv statements
+ = fmap (reannotS fst) $ transformUDStmt trans emptyExpEnv $ freevarsStmt statements
```

Comment:
yeah, about 30% less time/alloc. It could be better if I could minimise finding freevars in other heavy transforms such as nestBlocks as well.

Curiously now `foldStmt` is quite expensive, surely walking the Avalanche AST can't be that bad....